tests/test_client.py

```diff
@@ -578,3 +595,357 @@ def test_history_from_json():
+    await assert_no_schedules(client)
+
+    handle = await client.create_schedule(
+        f"schedule-{uuid.uuid4()}",
+        Schedule(
+            action=ScheduleActionStartWorkflow(
+                "kitchen_sink",
+                KSWorkflowParams(
+                    actions=[KSAction(result=KSResultAction("some result"))]
+                ),
+                id=f"workflow-{uuid.uuid4()}",
+                task_queue=worker.task_queue,
+            ),
+            spec=ScheduleSpec(calendars=[ScheduleCalendarSpec()]),
+            state=ScheduleState(paused=True),
+        ),
+    )
+    desc = await handle.describe()
+    assert desc.schedule.spec.calendars[0] == ScheduleCalendarSpec()
+    # Make sure that every next time has all zero time portion and is one day
+    # after the previous
+    assert len(desc.info.next_action_times) == 10
+    for i, time in enumerate(desc.info.next_action_times):
+        assert time.second == 0
+        assert time.minute == 0
+        assert time.hour == 0
+        if i > 0:
+            assert time == desc.info.next_action_times[i - 1] + timedelta(days=1)
+
+    await handle.delete()
+    await assert_no_schedules(client)
+
+
+async def test_schedule_trigger_immediately(
+    client: Client, worker: ExternalWorker, env: WorkflowEnvironment
+):
+    if env.supports_time_skipping:
+        pytest.skip("Java test server doesn't support schedules")
+    await assert_no_schedules(client)
+
+    # Create paused schedule that triggers immediately
+    handle = await client.create_schedule(
+        f"schedule-{uuid.uuid4()}",
+        Schedule(
+            action=ScheduleActionStartWorkflow(
+                "kitchen_sink",
+                KSWorkflowParams(
+                    actions=[KSAction(result=KSResultAction("some result"))]
+                ),
+                id=f"workflow-{uuid.uuid4()}",
+                task_queue=worker.task_queue,
+            ),
+            spec=ScheduleSpec(),
+            state=ScheduleState(paused=True),
+        ),
+        trigger_immediately=True,
+    )
+
+    # Confirm workflow result
+    desc = await handle.describe()
+    assert desc.info.num_actions == 1
+    action_exec = (await handle.describe()).info.recent_actions[0].action
+    assert isinstance(action_exec, ScheduleActionExecutionStartWorkflow)
+    assert (
+        "some result"
+        == await client.get_workflow_handle(
+            action_exec.workflow_id, run_id=action_exec.first_execution_run_id
+        ).result()
+    )
+
+    await handle.delete()
+    await assert_no_schedules(client)
+
+
+async def test_schedule_backfill(
+    client: Client, worker: ExternalWorker, env: WorkflowEnvironment
+):
+    if env.supports_time_skipping:
+        pytest.skip("Java test server doesn't support schedules")
+    await assert_no_schedules(client)
+
+    # Just in case it's on the minute boundary, move it off
+    now = datetime.utcnow()
```

Comment:
Didn't we decide this is not needed since the backfill start range is exclusive?

Reply:
Harmless in the test to be safe

Reply:
I'd argue its misleading if a user read this test, but not blocking this PR